### ISSUE TITLE
Invalid attributes value used for LDAP search, prevents login.

### ISSUE THREAD

**Issue by otopetrik** [heiglandreas/authLdap#35]
### Issue
After upgrade, unable to log in as a LDAP user.

ldap_search fails with $attributes (succeeds without $attributes).

failure tracked down to an invalid array:
$attributes start as
```
array("cn", "", "mail", "")
```
ldap_search called with a map:
```
array(0 => "cn", 2 => "mail")
```

### Proposed fix:

inside authLdap.php, function authLdap_login, line 229, instead of
```
$attributes = array_filter(array($authLDAPNameAttr, $authLDAPSecName, $authLDAPMailAttr, $authLDAPWebAttr));
```
it should probably read:
```
$attributes = array_values(array_filter(array($authLDAPNameAttr, $authLDAPSecName, $authLDAPMailAttr, $authLDAPWebAttr)));
```

**Comment by heiglandreas** [comment]
Thanks for pointing out. I'll dig into it later today. Expect a fix within 8 hours.

**Comment by heiglandreas** [comment]
This should be fixed with the newly available update to [1.4.3](https://github.com/heiglandreas/authLdap/releases/tag/1.4.3). Would you be so kind to test that and when everything works out just close this issue? Thanks!

**Comment by otopetrik** [comment]
Confirmed. Replaced 1.4.2+fix with 1.4.3 and it works. Thanks.
